### PR TITLE
feat(unlock-js) - add `transferFrom` to walletService

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
- add `transferFrom` to walletService, needed for https://github.com/unlock-protocol/unlock/issues/11836 

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by julien51 — COMMENTED** [review]
Please add tests here so we can make sure it works with our whole test suite!

File: packages/unlock-js/src/__tests__/integration/lock/transferFrom.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          global.suiteData)
+
+        keyOwner = accounts[0]
+        newOwner = accounts[6]
+
+        // purchase a new key
+        tokenId = await walletService.purchaseKey(
+          {
+            lockAddress,
+            owner: keyOwner,
```
Comment (julien51) [inline packages/unlock-js/src/__tests__/integration/lock/transferFrom.js:24]: Try removing ths line and see who the owner is?

File: packages/unlock-js/src/__tests__/integration/lock/transferFrom.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        const prevOwner = await web3Service.ownerOf(
+          lockAddress,
+          tokenId,
+          chainId
+        )
+        expect(prevOwner).toBe(keyOwner)
+
+        // transfer key to a new owner
+        await walletService.transferFrom({
+          from: keyOwner,
```
Comment (julien51) [inline packages/unlock-js/src/__tests__/integration/lock/transferFrom.js:60]: Use `prevOwner`

File: packages/unlock-js/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Changes
 
+# 0.38.0
```
Comment (julien51) [inline packages/unlock-js/CHANGELOG.md:3]: Please change package.json as well!

**Review by julien51 — COMMENTED** [review]
Finally! I do not see versions abive 10? Is that normal?

**Comment by kalidiagne** [comment]
> Finally! I do not see versions abive 10? Is that normal?

yes, is correct, we have this type of import

```
import v10 from '../v10'

export default {
  ...v10,
  totalKeys,
  grantKeyExtension,
  setEventHooks,
  setReferrerFee,
  version: 'v11',
  PublicLock: abis.PublicLock.v11,
}
```
we can also do the same for the other version in a different PR eventually

**Review by julien51 — APPROVED** [review]

**Pull request merged**